client/src/rpc_client.rs

```diff
@@ -128,16 +130,51 @@ impl RpcClient {
         Self::new_with_timeout(url, timeout)
     }
 
+    pub fn commitment(&self) -> CommitmentConfig {
+        self.commitment_config
+    }
+
+    fn use_deprecated_commitment(&self) -> Result<bool, RpcError> {
+        let use_deprecated_commitment = self.use_deprecated_commitment.read().unwrap();
+        if let Some(use_deprecated_commitment) = *use_deprecated_commitment {
+            Ok(use_deprecated_commitment)
+        } else {
+            drop(use_deprecated_commitment);
+            let cluster_version = self.get_version().map_err(|e| {
```

Comment:
It'd be nice to factor out the server version check so we only do one RPC call for commitment and tx encoding checks

Reply:
Ah, nice idea. That should simplify the write lock as well.

Reply:
Done in https://github.com/solana-labs/solana/pull/14797/commits/b39aa44cdaf2197b223804b85f2693fce6f55b6c